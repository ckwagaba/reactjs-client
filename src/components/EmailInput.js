import React, { Component } from 'react';

class EmailInput extends Component {
  render () {
    return (
      <div>
        <input type="email" className="text_input email_input" placeholder={this.props.placeholder} value={this.props.value} onChange={this.props.handleInput} />
      </div>
    );
  }
}

export default EmailInput;
