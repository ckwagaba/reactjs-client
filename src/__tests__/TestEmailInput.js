import React from 'react';
import { shallow } from 'enzyme';
import EmailInput from '../components/EmailInput';

describe('EmailInput component', () => {

  const component = shallow(
    <EmailInput />
  );

  it('should exist', () => {
    expect(component).toMatchSnapshot();
  });

});
