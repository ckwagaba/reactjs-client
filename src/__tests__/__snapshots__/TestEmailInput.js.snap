// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailInput component should exist 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmailInput />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        className="text_input email_input"
        onChange={undefined}
        placeholder={undefined}
        type="email"
        value={undefined}
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "text_input email_input",
        "onChange": undefined,
        "placeholder": undefined,
        "type": "email",
        "value": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          className="text_input email_input"
          onChange={undefined}
          placeholder={undefined}
          type="email"
          value={undefined}
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "text_input email_input",
          "onChange": undefined,
          "placeholder": undefined,
          "type": "email",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
